mantidproject/mantid
# Fix importing string series logs

Importing string series logs into MantidPlot (via Sample Logs dialog) has been broken since Mantid 3.7, but worked in Mantid 3.6 and before. Re-enable this functionality by setting the column type correctly.

Also matched the old formatting by left-aligning/vertically centering text and making the column wide enough for its contents.

**To test:**

- Create a dataset with a string series log:
```python
ws = CreateSampleWorkspace()
run = ws.mutableRun()
prop = StringTimeSeriesProperty("string_series")
prop.addValue("2016-11-10T11:58:20", "hello")
prop.addValue("2016-11-10T11:58:24", "message2")
prop.addValue("2016-11-10T11:58:26", "message3")
run.addProperty("string_series", prop, False)
```
- Right-click on the workspace in MantidPlot and select *Sample Logs...*
- In the dialog, select the `string_series` log and click *Import selected log*
- Log should be imported correctly, with values in the table - not a blank values column. (See the issue)
- Can also try this with a real workspace that has a string series log. (I used the `ICPEVENT` log from `EMU00066122.nxs`)

Fixes #17923.

[Release notes](https://github.com/mantidproject/mantid/blob/bd01f708ebaf7e35d6d1e86260e2d6d3b71e42cc/docs/source/release/v3.9.0/ui.rst#bugs-resolved) have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
